en/02_Developer_Guides/00_Model/04_Data_Types_and_Casting.md

```diff
@@ -103,6 +104,65 @@ class Car extends DataObject
+
+Some good use cases for generated columns include:
+
+- sorting in a gridfield on a complex summary field: It's common to use a getter method to get some value derived from your database fields (e.g. a discounted price), and use that in `summary_fields`. With a generated column, you can remove the getter method and the [`GridFieldSortableHeader`](api:SilverStripe\Forms\GridField\GridFieldSortableHeader) component will be able to sort using the column.
+- using functional indexes: generated columns can be included in indexes, allowing you to sort or filter by complex expressions in an efficient way.
+- data integrity: unlike generating values inside an `onBeforeWrite()` method, generated column values will be correct even if you update the record with raw SQL. You also don't need to manually update values for historic records (e.g. when using [versioning](https://docs.silverstripe.org/en/6/developer_guides/model/versioning/#versioning)) even if you change the logic that determines the value.
+- reduce repetition: instead of using a complex expression in multiple different places, you can just give the expression a name with a generated column.
+
+The format for adding a generated column is `Generated("<datatype>", "<expression>", "<STORED|VIRTUAL>")`. Let's break that down:
+
+|argument|explanation|example|
+|---|---|---|
+|`<datatype>`|The injector specification for a `DBField` instance that represents your generated field in the database and when getting its value, etc.|`Varchar(255)`, `Boolean`, etc|
+|`<expression>`|The SQL expression used to generate values for this field. Best practice is to use ANSI quotes around column names. You need to add more escape characters as you might expect, e.g. to represent a FQCN in a string literal you will need something like `\'App\\\\\\\\Model\\\\\\\\Car\'`.|`\\"Price\\" * (1.0 - \\"Discount\\")`|
+| `<STORED\|VIRTUAL>` | Whether the value is calculated when the record is updated and then stored in the database (use `STORED`), or calculated only when your query includes it (use `VIRTUAL`). |`STORED`|
+
+> [!TIP]
+> For lengthy or complex expressions, it is often best to use a [nowdoc](https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.nowdoc) multi-line string to make the expression easier to read.
+>
+> A nowdoc will also treat backslashes literally, which can be especially useful when dealing with FQCN in the expression.
+
+If you had a car model and wanted to calculate a discounted price using a generated column, you could do something like this:
+
+```php
+namespace App\Model;
+
+use SilverStripe\ORM\DataObject;
+
+class Car extends DataObject
+{
+    private static $db = [
+        'Condition' => 'Enum("New,Fair,Junk", "Fair")',
+        'Price' => 'Currency',
+        // Discount the car for a fixed percentage based on the condition of the car
+        'Discount' => <<<'SPEC'
```

Comment:
See https://github.com/silverstripe/silverstripe-assets/pull/694#discussion_r2208956301 for why I used `SPEC` as the delimiter.